### ISSUE TITLE
Implement scbasset

### ISSUE THREAD

**Issue by adamgayoso** [scverse/scvi-tools#1729]
This meta-issue is for tracking the implementation of scbasset

- [ ] #1730 
- [ ] Implement module and basic model
- [ ] Add more model features

**Comment by macwiatrak** [comment]
Is anybody actively working on this? If not, happy to contribute.

**Comment by adamgayoso** [comment]
@macwiatrak please go ahead. One caveat about scbasset is that minibatching happens on the regions axis, which is not something we have implemented in scvi tools yet.

A quick and easy solution is to just transpose the anndata object, but then this creates an issue with how to get the batch id information into the model. #1810 helps get it started, but also duplicates a lot of code with obs/obsm that directly mirrors var/varm.

**Comment by dagarfield** [comment]
Would be a big fan of this as a feature....

**Comment by macwiatrak** [comment]
I have implemented and tested the model architecture, but I have not yet started on the anndata object (which is the bulk of the work). I could come back to it earliest after a couple of weeks, so if anybody else would like to get started on it feel free.

**Comment by adamgayoso** [comment]
are you able to share the model architecture?

**Comment by macwiatrak** [comment]
Sure @adamgayoso, I ported it into a separate repo - https://github.com/macwiatrak/scBasset-pytorch. Heads up it's a bare model architecture, not implemented in the `scvi-tools` format.

**Comment by adamgayoso** [comment]
I think we can handle that part

**Comment by adamgayoso** [comment]
@macwiatrak @gokceneraslan I have a working version in the linked PR. I think there are still minor but important architecture differences (initializations, batch norm hyperparams, exact architecture hyperparams). As well as no batch correction part (but the data is hooked in there)

**Comment by macwiatrak** [comment]
I can have a look at the above after I come back from holidays (16.01). Feel free to get started on it if it's something urgent.
